R/inputs-spectrum.R

```diff
@@ -284,26 +329,43 @@ read_dp_art_dec31 <- function(dp) {
   ##
   ## * Implemented same as adult adjustment factor above
 
-  if (exists_dptag("<ChildARTAdjFactorFlag>") &&
-        dpsub("<ChildARTAdjFactorFlag>", 2, 4) == 1) {
+  ## Initialise
+  child_artadj_factor <- rep(1.0, length(child_art))
+  child_artadj_absolute <- rep(0.0, length(child_art))
+
+  ## Flag to use adjustment
+  use_child_artadj <- exists_dptag("<ChildARTAdjFactor MV>") &&
+    (!exists_dptag("<ChildARTAdjFactorFlag>") ||
+       (exists_dptag("<ChildARTAdjFactorFlag>") &&
+          dpsub("<ChildARTAdjFactorFlag>", 2, 4) == 1))
+
+  if (use_child_artadj) {
 
     child_artadj_factor <- as.numeric(dpsub("<ChildARTAdjFactor MV>", 2, timedat.idx))
 
+    if(exists_dptag("<ChildPatsAllocToFromOtherRegion MV>")) {
+      child_artadj_absolute <- as.numeric(dpsub("<ChildPatsAllocToFromOtherRegion MV>", 2, timedat.idx))
+    }
+
     ## Only apply if is number (! is percentage)
     child_artadj_factor <- child_artadj_factor ^ !child_art_isperc
+    child_artadj_absolute <- child_artadj_absolute ^ !child_art_isperc
 
+    ## First add absolute adjustment, then apply scalar adjustment (Spectrum procedure)
+    child_art <- child_art + child_artadj_absolute
     child_art <- child_art * child_artadj_factor
```

Comment:
In contrast to above where the adjusted was written into a new variable (`art15plus_num_adj`), here it looks like `child_art` is being overwritten by the adjusted value.

I would have expected these should be consistent?

Reply:
Updated to be consistent with adult calculations